### PR TITLE
django initial setup + mysql integration

### PR DESCRIPTION
**Pull request by prakharchoudhary**
Hello @RishabhJain2018  sir, this is the initial django setup. Please check the DATABASE section in settings.py once and tell me if it needs any corrections.

### CODE REVIEW

File: Q_hub/Q_hub/config_keys.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+MYSQL_USERNAME = ""
+
+MYSQL_PASSWORD = ""
+
+DATABASE_NAME = ""
```
Comment (RishabhJain2018) [inline Q_hub/Q_hub/config_keys.py]: Please leave a blank link here.

File: Q_hub/Q_hub/settings.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+USE_TZ = True
+
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/1.9/howto/static-files/
+
+STATIC_URL = '/static/'
+STATICFILES_DIRS = (
+    os.path.join(BASE_DIR, "static"),
+)
```
Comment (RishabhJain2018) [inline Q_hub/Q_hub/settings.py]: a blank line here also.

**Review by RishabhJain2018 — COMMENTED** [review]
@prakharchoudhary Yeah..I think the Database config is correct, but do check on your local machine whether it is working or not.

**Comment by RishabhJain2018** [comment]
@prakharchoudhary Open the PR according to the issues. You can also open the issues for different PR.

**Comment by prakharchoudhary** [comment]
@RishabhJain2018 sir i have changed the requirements.txt on my branch. I am trying to create a new pull request however it only shows the 'View pull request' for previous PR.
screenshot 15

**Comment by RishabhJain2018** [comment]
just make a new branch from master & then modify the `requirements.txt` file and then push that branch.

File: .travis.yml [hunk 3]
```diff
@@ -8,5 +8,13 @@ install:
   - pip install -r requirements.txt
```
Comment (RishabhJain2018) [inline .travis.yml:8]: Remove this file from here as it is already there in the repo.

File: Q_hub/app/admin.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from django.contrib import admin
+from models import *
+# Register your models here.
+
+admin.site.register(SubjectCode)
+admin.site.register(QuestionDetail)
```
Comment (RishabhJain2018) [inline Q_hub/app/admin.py]: Add a blank line here.

File: Q_hub/app/forms.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from django.contrib.auth.models import User
+from django import forms
+
+class LoginForm(forms.Form):
+	username = forms.CharField(label='username', max_length=30, widget = forms.TextInput(attrs={'placeholder': 'Username'}))
+	password = forms.CharField(
+		label='password', 
+		widget=forms.PasswordInput(attrs={'placeholder': 'Password'})
+		)
```
Comment (RishabhJain2018) [inline Q_hub/app/forms.py]: Blank line here also.

File: Q_hub/app/models.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+class QuestionDetail(models.Model):
+	owner = models.ForeignKey(User)
+	date_created = models.DateTimeField(auto_now_add=True)
+	question = models.TextField()
+	marks = models.IntegerField()
+	unit = models.IntegerField()
+	n_used = models.IntegerField()
+	imp = models.BooleanField(default=False)
+
```
Comment (RishabhJain2018) [inline Q_hub/app/models.py]: Blank line here also.

File: Q_hub/static/style.css [hunk 7]
```diff
@@ -0,0 +1,83 @@
+nav{
```
Comment (RishabhJain2018) [inline Q_hub/static/style.css:1]: Make a different PR for the frontend related files.

File: Q_hub/static/style.css [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    text-align:left;
+   }
+footer
+{
+	 padding:8px;
+     color: white;
+    background-color: #000040;
+    clear: left;
+    text-align: center;
+}
```
Comment (RishabhJain2018) [inline Q_hub/static/style.css]: A new line here.

File: Q_hub/templates/base.html [hunk 9]
```diff
@@ -0,0 +1,27 @@
+{% load staticfiles %}
```
Comment (RishabhJain2018) [inline Q_hub/templates/base.html:1]: Put the html files in a different file.

File: requirements.txt [hunk 10]
```diff
@@ -1 +1,2 @@
-django=1.9.2
\ No newline at end of file
+django==1.9.2
```
Comment (RishabhJain2018) [inline requirements.txt]: remove this file.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
Put all the .pyc files in gitignore.

File: Q_hub/app/admin.py (hunk 4 above)
Comment (prakharchoudhary) [reply Q_hub/app/admin.py]: a blank line at the end of code?

Comment (RishabhJain2018) [reply Q_hub/app/admin.py]: yes.

File: Q_hub/static/style.css (hunk 7 above)
Comment (prakharchoudhary) [reply Q_hub/static/style.css:1]: Sir do you mean I should exclude the templates and static folder while making the PR? because if you do then i will have to look for how to do it

Comment (RishabhJain2018) [reply Q_hub/static/style.css:1]: yeah go according to the issue. It says` django initial setup`. So it would be good not to include frontend related files in this PR. For frontend related files open up a different issue & then create a PR. Let me know if you have queries.

Comment (prakharchoudhary) [reply Q_hub/static/style.css:1]: Sir I looked up for it but couldn't find anything. How can I exclude only the frontend files while making a PR? Does a PR not mean that the VCS compares the master and remote branch and integrates the latest changes of the remote branch into the master. How do frontend files affect the merging process?

Comment (RishabhJain2018) [reply Q_hub/static/style.css:1]: Well, you don't have to look for anything , it's just the common sense that you could have used for including only the backend files. Don't use `git add .` for adding the files instead add the backend files manually & there are hardly 5-6 files. Moreover, frontend files don't affect the merging process, its just to make the merge commits clean and specific to the issue, so that in future if we have to revert then we can easily revert back.
